packages/next/build/webpack/loaders/next-image-loader.js

```diff
@@ -0,0 +1,58 @@
+import loaderUtils from 'next/dist/compiled/loader-utils'
+import sizeOf from 'image-size'
+import { processBuffer } from '../../../next-server/server/lib/squoosh/main'
+import fs from 'fs'
+
+async function nextImageLoader(content) {
+  this.cacheable && this.cacheable(true)
+  this.addDependency(this.resourcePath)
+
+  const context = this.rootContext
+  const opts = { context, content }
+  const interpolatedName = loaderUtils.interpolateName(
+    this,
+    '/[path][name].[ext]',
+    opts
+  )
+
+  let extension = loaderUtils.interpolateName(this, '[ext]', opts)
+  if (extension === 'jpg') {
+    extension = 'jpeg'
+  }
+
+  if (interpolatedName.slice(0, 7) !== '/public') {
+    const err = new Error(
+      'Static Image loader used with filepath not in the /public directory: ' +
+        interpolatedName
+    )
+    this.emitError(err)
+    return
+  }
+
+  const src = interpolatedName.slice(7)
+  const imageSize = sizeOf(this.resourcePath)
+  let dataURI
+  if (extension === 'jpeg' || extension === 'png') {
+    const fileBuffer = Buffer.from(fs.readFileSync(this.resourcePath))
+    const resizedImage = await processBuffer(
+      fileBuffer,
+      [{ type: 'resize', width: 6 }],
+      extension,
+      0
+    )
+    dataURI = `data:image/${extension};base64,${resizedImage.toString(
+      'base64'
+    )}`
+  }
+
+  const stringifiedData = JSON.stringify({
+    src,
+    height: imageSize.height,
+    width: imageSize.width,
+    dataURI,
+  })
+
+  return `${'export default '} ${stringifiedData};`
```

Comment:
I would prefer to put the meta information into a separate export, so the default export stays only the URL for backward-compat reasons. Currently enabling this feature, breaks all existing image imports as it returns an object instead of an url now.

You would use it like that:

``` js
import * as testImg from '../public/foo/test-rect.jpg'

      <Image
        id="basic-static"
        src={testImg}
        layout="fixed"
        placeholder="blur"
      />
```

@timneutkens What do you think?

The import might be easy to get wrong... A lint rule could help with that.

Reply:
> Currently enabling this feature, breaks all existing image imports as it returns an object instead of an url now.

Image imports are not supported in Next.js currently, so that should be fine, potentially we can do the same as the css imports detection to disable the loader if there's customization that would break? 🤔